### DISCUSSION TITLE
I don't have access to hooks using hooks_riverpod

### DISCUSSION THREAD

**Discussion by JanHitako** [rrousselGit/riverpod#2666] [Q&A] [answered]
Hi, beginner question here.

I have been learning and using hooks for a while. Now, I want to learn to use riverpod.

on a new flutter project on vscode, I do : dart : add dependency : hooks_riverpod

on the main.dart file, I import 'package:hooks_riverpod/hooks_riverpod.dart';

Then I replace the default stateful widget with HookConsumerWidget instead of HookWidget

In the build, I then try to add some hooks, but none are recongnized or can be automatically imported.
Any hooks I try t use give therefore similar errors to this:

> The method 'useAnimationController' isn't defined for the type '_MyHomePageState'.
> Try correcting the name to the name of an existing method, or defining a method named 'useAnimationController'

Why is it? What am I doing wrong here? I tried searching online but I have found no similar question at all so I am guessing there is a mistep on my part as a beginner?

Thanks!

**Comment by JanHitako** [comment]
Fixed it by modifying hook_riverpod.dart from this

```
// ignore: invalid_export_of_internal_element
export 'package:flutter_riverpod/flutter_riverpod.dart';

export 'src/consumer.dart';
```

to this:

```
// ignore: invalid_export_of_internal_element
export 'package:flutter_riverpod/flutter_riverpod.dart';

export 'package:flutter_hooks/flutter_hooks.dart';

export 'src/consumer.dart';
```

I'm not sure why hooks_riverpod only export riverpod but not hooks so I'm not sure either if this is the correct path to fix the issue but it works for me as of now.

Would like some input regarding this from anyone having more experience with the package, thanks!

**Comment by rrousselGit** [comment, marked as answer]
flutter_hooks is installed and imported separately. This ensures better package versioning, in case flutter_hooks releases some breaking change & co.

**Reply by JanHitako** [reply]
This actually makes sense when I think about it. This also avoid the sort of mess that is introduced by packages like GetX. 
Thanks for the quick reply Remi!
